src/Data/Text/Builder/Linear/Core.hs

```diff
@@ -289,6 +342,34 @@ prependBounded maxSrcLen prepender appender (Buffer (Text dst dstOff dstLen))
       pure $ Text new newOff (dstLen + srcLen)
 {-# INLINE prependBounded #-}
 
+-- | Low-level routine to prepend data of unknown size to a 'Buffer'.
+--
+-- Contrary to 'prependBounded', only use a prepend action.
+prependBounded'
```

Comment:
_Internal_ new helper. If deemed a useful addition to the API, maybe we need a better name.

Useful function when the prepender is always faster that the appender.